test/createService.spec.js

```diff
@@ -0,0 +1,48 @@
+/* global describe, it */
+import chai, { expect } from 'chai';
+import sinon from 'sinon';
+import sinonChai from 'sinon-chai';
+import createService from '../src/createService';
+chai.use(sinonChai);
+
+describe('createService', () => {
+  const mySpec = {
+    initialize: sinon.stub(),
+    customFunction1() { return 'value1'; },
+    customFunction2() { return 'value2'; },
+    returnsThis() { return this; }
+  };
+
+  const myOptions = {
+    app: {}
+  };
+
+  const MyService = createService(mySpec);
+  const myServiceInstance = new MyService(myOptions);
+
+  it('throws error when an app instance is not provided', () => {
+    expect(() => new MyService()).to.throw(Error, 'App instance not provided.');
+  });
+
+  it('creates instance from my service', () => {
+    expect(myServiceInstance).to.be.instanceOf(MyService);
+  });
+
+  it('executes the initialize() at construction', () => {
+    expect(mySpec.initialize).to.be.callCount(1);
```

Comment:
using `chai-spies`:
`expect(mySpec.initialize).to.have.been.called.once`
what do you think?